modules/web_hosting/main.tf

```diff
@@ -0,0 +1,550 @@
+    time_sleep.wait_120_seconds,
+    google_compute_router_nat.nat_gw_region_primary
+    ]
+}
+ 
+resource "google_compute_instance" "web3_vpc_xlb" {
+  project      = local.project.project_id
+  zone         = "${var.region}-c"
+  name         = "web3-vpc-xlb"
+  machine_type = "f1-micro"
+  allow_stopping_for_update = true
+  metadata_startup_script   = data.template_file.sample_webapp_metadata_startup_script.rendered
+  metadata = {
+    enable-oslogin = true
+  }
+  boot_disk {
+    initialize_params {
+      image = "debian-11-bullseye-v20220822"
+    }
+  }
+ 
+  network_interface {
+    subnetwork         = google_compute_subnetwork.subnetwork_primary.name
+    subnetwork_project = local.project.project_id
+    network_ip         = cidrhost(google_compute_subnetwork.subnetwork_primary.ip_cidr_range, 4)
+    # access_config {
+    #   // Ephemeral IP
+    # }
+  }
+  service_account {
+    # Google recommends custom service accounts that have cloud-platform scope and permissions granted via IAM Roles.
+    email  = google_service_account.sa_p_cloud_sql.email
+    scopes = ["cloud-platform"]
+  }
+  
+  depends_on = [
+    time_sleep.wait_120_seconds,
+    google_compute_router_nat.nat_gw_region_primary
+    ]
+}
+ 
+resource "google_compute_instance" "web4_vpc_xlb" {
+  project      = local.project.project_id
+  zone         = "asia-south1-c"
+  name         = "web4-vpc-xlb"
+  machine_type = "f1-micro"
+  allow_stopping_for_update = true
+  metadata_startup_script   = data.template_file.sample_app_metadata_startup_script.rendered
+  metadata = {
+    enable-oslogin = true
+  }
+  boot_disk {
+    initialize_params {
+      image = "debian-11-bullseye-v20220822"
+    }
+  }
+ 
+  network_interface {
+    subnetwork         = google_compute_subnetwork.subnetwork_secondary.name
+    subnetwork_project = local.project.project_id
+    network_ip         = cidrhost(google_compute_subnetwork.subnetwork_secondary.ip_cidr_range, 2)
+    # access_config {
+    #   // Ephemeral IP
+    # }
+  }
+  service_account {
+    # Google recommends custom service accounts that have cloud-platform scope and permissions granted via IAM Roles.
+    email  = google_service_account.sa_p_cloud_sql.email
+    scopes = ["cloud-platform"]
+  }
+
+  depends_on = [
+    time_sleep.wait_120_seconds,
+    google_compute_router_nat.nat_gw_region_secondary
+    ]
+}
+
+
+#########################################################################
+# GCS bucket / Bucket Objects / Bucket Bindings
+#########################################################################
+
+
+resource "google_storage_bucket" "gcs_image_bucket" {
+  name          = join("",["gcs_image_bucket-",local.project.project_id])
+  location      = "US"
+  project       = local.project.project_id
+  force_destroy = true
+  depends_on    = [
+    google_project_service.enabled_services,
+    time_sleep.wait_120_seconds]
+}
+
+resource "google_storage_object_access_control" "public_rule" {
```

Comment:
In general we don't recommend using ACLs on individual bucket objects.  Is there a reason why the general public requires access to this image?  Can't we serve it through the web hosting backend?

Reply:
We want to showcase the GCS as the backend functionality behind Ext. LB.